lib/dev_server.dart

```diff
@@ -0,0 +1,131 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:mime/mime.dart' as mime;
+import 'package:path/path.dart' as pathlib;
+
+import 'butterfly.dart';
+
+/// Butterfly development server that serves the application.
+class ButterflyDevServer {
+  static const String _devChannelPath = '/__butterfly_dev_channel__';
+
+  /// Starts a development server listening on the given [port] number.
+  static Future<ButterflyDevServer> start(int port) async {
+    final server = await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, port);
+    print('Butterfly dev server listening on http://localhost:$port');
```

Comment:
Totally should, but I'll postpone it to move faster: https://github.com/yjbanov/butterfly/issues/12